File: src/types/feed-item.ts
export interface FeedItem {
  id: string;
  source_id: string;
  external_id: string;
  title: string;
  thumbnail_url: string | null;
  published_at: string | null;
  duration: number | null;
  download_status: "not_downloaded" | "downloading" | "downloaded" | "error";
  warehouse_item_id: string | null;
  metadata_complete: boolean;
  created_at: string;
}

export interface CreateFeedItemRequest {
  source_id: string;
  external_id: string;
  title: string;
  thumbnail_url?: string | null;
  published_at?: string | null;
  duration?: number | null;
}

export interface UpdateFeedItemRequest {
  download_status?: "not_downloaded" | "downloading" | "downloaded" | "error";
  warehouse_item_id?: string | null;
}

export interface FeedItemCounts {
  total: number;
  downloaded: number;
  not_downloaded: number;
}

export interface SyncEvent {
  source_id: string;
  status: "started" | "completed" | "error";
  message: string | null;
  new_items: number | null;
}

export interface MetadataEvent {
  feed_item_id: string;
  status: "started" | "completed" | "error";
  message?: string;
}

File: src-tauri/src/models/feed_item.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FeedItem {
    pub id: String,
    pub source_id: String,
    pub external_id: String,
    pub title: String,
    pub thumbnail_url: Option<String>,
    pub published_at: Option<String>,
    pub duration: Option<i64>,
    pub download_status: String,
    pub warehouse_item_id: Option<String>,
    pub metadata_complete: bool,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateFeedItemRequest {
    pub source_id: String,
    pub external_id: String,
    pub title: String,
    pub thumbnail_url: Option<String>,
    pub published_at: Option<String>,
    pub duration: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateFeedItemRequest {
    pub download_status: Option<String>,
    pub warehouse_item_id: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct FeedItemCounts {
    pub total: i32,
    pub downloaded: i32,
    pub not_downloaded: i32,
}
